internal/schemautil/userconfig/apiconvert/toapi.go

```diff
@@ -176,10 +177,14 @@ func itemToAPI(
 	//
 	// We check that there are more than three elements in the fk slice, because we don't want to send the value if
 	// the parent object is the root object.
-	_, e := d.GetOk(fks)
+	if o && len(fk) > 3 {
+		match := regexp.MustCompile(`\.\d($|\.)`).FindAllStringIndex(fks, -1)
 
-	if o && e && len(fk) > 3 && d.HasChange(fks[:strings.LastIndex(fks, ".0")]) {
-		o = false
+		_, e := d.GetOk(fks)
+
+		if match != nil && (e || !reflect.ValueOf(v).IsZero()) && d.HasChange(fks[:match[len(match)-1][0]]) {
```

Comment:
We can benefit from having some of the comments, and it needs to be clarified at first glance what we are doing here.

Reply:
there's actually a comment already above this whole line, but sure we could make this clearer! I'll update the PR